packages/filecoin-api/src/storefront/api.ts

```diff
@@ -189,3 +189,10 @@ export interface PieceOfferMessage {
 export interface DataAggregationProofNotFound extends Failure {
   name: 'DataAggregationProofNotFound'
 }
+
+export interface ContentStore<RecKey, Rec> {
+  /**
+   * Gets a record from the store.
+   */
+  stream: (key: RecKey) => Promise<Result<AsyncIterable<Rec>, StoreGetError>>
```

Comment:
Not sure if useful, but `ReadableStream` _is_ async iterable and you get extra utility for controlling flow and pipelining. Also developer familiarity with https://developer.mozilla.org/en-US/docs/Web/API/Blob/stream

```suggestion
  stream: (key: RecKey) => Promise<Result<ReadableStream<Rec>, StoreGetError>>
```